subiquitycore/core.py

```diff
@@ -121,22 +142,16 @@ def run(self):
                 self.common['loop'].event_loop))
 
         try:
-            self.set_alarm_in(0.05, self.welcome)
-            for k in self.controllers.keys():
+            self.set_alarm_in(0.05, self.next_screen)
+            for k in self.common['controllers'].keys():
```

Comment:
Of course, if `self.common['controllers']` is a dictionary, you don't need `.keys()` since that's the default iteration strategy for dictionaries.